src/main/kotlin/lotto/dto/LottoResult.kt

```diff
@@ -0,0 +1,22 @@
+package lotto.dto
+
+import lotto.entity.Rank
+
+class LottoResult private constructor(
+    val rankToHowManyWins: Map<Rank, Int>,
+    val rateOfReturn: Double,
+) {
+    companion object {
+        fun of(rankToHowManyWins: Map<Rank, Int>, purchaseCost: Long): LottoResult {
+            return LottoResult(
+                rankToHowManyWins = rankToHowManyWins,
+                rateOfReturn = calculateRateOfReturn(rankToHowManyWins.calculateTotalPrize(), purchaseCost.toDouble())
+            )
+        }
+
+        private fun Map<Rank, Int>.calculateTotalPrize(): Double =
+            this.entries.sumOf { (rank, howManyWins) -> howManyWins * rank.prize }.toDouble()
```

Comment:
가급적 method chain의 경우, 한 줄의 하나의 메서드만 호출하는 습관을 들여보시면 어떨까요? 😃

```suggestion
            this.entries.sumOf { (rank, howManyWins) -> howManyWins * rank.prize }
                .toDouble()
```